### ISSUE TITLE
AltAz -> ITRS

### ISSUE THREAD

**Issue by svr93** [astropy/astropy#4249]
When I try to convert AltAz coordinates with distance, I get
```python
from astropy import units as u
from astropy.time import Time
from astropy.coordinates import SkyCoord, EarthLocation, AltAz
n = EarthLocation(lat=41.3*u.deg, lon=-74*u.deg, height=390*u.m)
time = Time('2012-7-12 23:00:00')
c = SkyCoord(alt=37*u.deg,az=25*u.deg,distance=7800*u.km,obstime=time,location=n,frame='altaz')
c
<SkyCoord (AltAz: obstime=2012-07-12 23:00:00.000, location=(1322772.7367951265, -4613056.748218293, 4187767.9533615136) m, pressure=0.0 hPa, temperature=0.0 deg_C, relative_humidity=0, obswl=1.0 micron): (az, alt, distance) in (deg, deg, km)
    (25.0, 37.0, 7800.0)>
c.transform_to('itrs')
<SkyCoord (ITRS: obstime=2012-07-12 23:00:00.000): (x, y, z) in km
    (139268182.96865937, 51622160.98614869, 412898505.580033)>
```
Why calculated coordinates so high? I have similar problem in case of test converting AltAz -> AltAz
```python
c.transform_to('altaz')
<SkyCoord (AltAz: obstime=2012-07-12 23:00:00.000, location=(1322772.7367951265, -4613056.748218293, 4187767.9533615136) m, pressure=0.0 hPa, temperature=0.0 deg_C, relative_humidity=0, obswl=1.0 micron): (az, alt, distance) in (deg, deg, m)
    (25.0, 37.0, 501344127124.25921631)>
c.transform_to('altaz').transform_to('altaz')
<SkyCoord (AltAz: obstime=2012-07-12 23:00:00.000, location=(1322772.7367951265, -4613056.748218293, 4187767.9533615136) m, pressure=0.0 hPa, temperature=0.0 deg_C, relative_humidity=0, obswl=1.0 micron): (az, alt, distance) in (deg, deg, m)
    (25.0, 37.0, 566987610424.75195312)>
```

**Comment by embray** [comment]
Hi @svr93, instead of screenshots can you please just copy and paste the text from your terminal?  On GitHub you can place it between triple-backticks to keep it formatted as plain text as shown here: https://help.github.com/articles/github-flavored-markdown/#fenced-code-blocks

**Comment by svr93** [comment]
Hi @embray, I have just replaced screenshots, sorry for bad-style

**Comment by embray** [comment]
No problem, it's just easy to read, and to also make sure we're copying your examples verbatim.

Sorry I don't have an answer to the actual question.  Will have to wait for someone with more expertise to have a look.  I *think* this might be an already known issue that's being fixed.  Something about it looks very familiar--like, I think maybe some quantity gets incorrectly changed from meters to kilometers.  Not sure though.

**Comment by embray** [comment]
Oh, I thought this seemed familiar--this is *probably* a duplicate of #4221.  Let me see if it's helped by #4254.

**Comment by embray** [comment]
I confirmed that this example seems to be fixed by #4254, so I'm marking this for now as a duplicate of #4221 that can be closed once the PR is merged.

Granted, I didn't check that the coordinate transformation is actually *correct*, but it at least looks like more sensible numbers:
```python
In [4]: n = EarthLocation(lat=41.3*u.deg, lon=-74*u.deg, height=390*u.m)

In [5]: time = Time('2012-7-12 23:00:00')

In [6]: c = SkyCoord(alt=37*u.deg,az=25*u.deg,distance=7800*u.km,obstime=time,location=n,frame='altaz')

In [7]: c
Out[7]: 
<SkyCoord (AltAz: obstime=2012-07-12 23:00:00.000, location=(1322772.7367951267, -4613056.748218293, 4187767.9533615136) m, pressure=0.0 hPa, temperature=0.0 deg_C, relative_humidity=0, obswl=1.0 micron): (az, alt, distance) in (deg, deg, km)
    (25.0, 37.0, 7800.0)>

In [8]: c.transform_to('itrs')
/internal/1/root/home/embray/src/astropy/astropy/astropy/coordinates/builtin_frames/cirs_observed_transforms.py:117: RuntimeWarning: invalid value encountered in arcsin
  distance = np.sin(delta + np.arcsin(r*sindoverd))/sindoverd
Out[8]: 
<SkyCoord (ITRS: obstime=2012-07-12 23:00:00.000): (x, y, z) in km
    (3085.2482689, 1143.60063745, 9147.05981236)>
```

**Comment by eteq** [comment]
Yep, I have every reason to believe this is exactly what #4254 was supposed to fix.

**Comment by eteq** [comment]
I'm going to go ahead and close this given that #4254 is now merged.  But if you think it still isn't fixed, @svr93 , feel free to re-open!

**Comment by svr93** [comment]
Hello!
I found time for new check. You can see results below
```python
from astropy import units as u
from astropy.time import Time
from astropy.coordinates import SkyCoord, EarthLocation, AltAz
location = EarthLocation(lat=0*u.deg, lon=0*u.deg, height=0*u.m)
time = Time('2012-7-12 23:00:00')
coord = SkyCoord(az=0*u.deg,alt=90*u.deg,distance=5*u.km,location=location,obstime=time,frame='altaz')
coord
<SkyCoord (AltAz: obstime=2012-07-12 23:00:00.000, location=(6378137.0, 0.0, 0.0) m, pressure=0.0 hPa, temperature=0.0 deg_C, relative_humidity=0, obswl=1.0 micron): (az, alt, distance) in (deg, deg, km)
    (0.0, 90.0, 5.0)>
coord.transform_to('altaz')
/usr/local/lib/python2.7/site-packages/astropy/coordinates/builtin_frames/cirs_observed_transforms.py:117: RuntimeWarning: invalid value encountered in arcsin
  distance = np.sin(delta + np.arcsin(r*sindoverd))/sindoverd
<SkyCoord (AltAz: obstime=2012-07-12 23:00:00.000, location=(6378137.0, 0.0, 0.0) m, pressure=0.0 hPa, temperature=0.0 deg_C, relative_humidity=0, obswl=1.0 micron): (az, alt, distance) in (deg, deg, m)
    (270.31046215, 90.0, 5000.0)>
```
Not good value for first degree value, but it's boundary point, so nothing bad
```python
coord.transform_to('itrs')
<SkyCoord (ITRS: obstime=2012-07-12 23:00:00.000): (x, y, z) in km
    (6383.13367696, 0.18226191, 0.00000038)>
```
All ok) But
```python
coord = SkyCoord(az=0*u.deg,alt=85*u.deg,distance=5*u.km,location=location,obstime=time,frame='altaz')
>>> coord.transform_to('altaz')
/usr/local/lib/python2.7/site-packages/astropy/units/quantity.py:477: RuntimeWarning: invalid value encountered in arcsin
  out = function(inputs[0], obj_array)
<SkyCoord (AltAz: obstime=2012-07-12 23:00:00.000, location=(6378137.0, 0.0, 0.0) m, pressure=0.0 hPa, temperature=0.0 deg_C, relative_humidity=0, obswl=1.0 micron): (az, alt, distance) in (deg, deg, m)
    (0.0, 85.0, nan)>
```
I only changed alt value! See more:
```python
coord = SkyCoord(az=1*u.deg,alt=85*u.deg,distance=5*u.km,location=location,obstime=time,frame='altaz')
coord.transform_to('altaz')
<SkyCoord (AltAz: obstime=2012-07-12 23:00:00.000, location=(6378137.0, 0.0, 0.0) m, pressure=0.0 hPa, temperature=0.0 deg_C, relative_humidity=0, obswl=1.0 micron): (az, alt, distance) in (deg, deg, m)
    (1.0, 85.0, nan)>
coord.transform_to('itrs')
<SkyCoord (ITRS: obstime=2012-07-12 23:00:00.000): (x, y, z) in km
    (nan, nan, nan)>
```
